src/Header.test.js

```diff
@@ -0,0 +1,43 @@
+import { cleanup, render } from '@testing-library/react';
+// import { renderer } from '@testing-library/react';
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import Header from './components/Header';
+
+afterEach(cleanup);
+it('containts search link', async () => {
+  const { getByTestId } = render(
+    <Router>
```

Comment:
Here it's better to render the App component instead of the Header. Then you can just write this

```
render(<App />)
```

Alternatively you can wrap the Header additionally in a ThemeProvider

```
<ThemeProvider theme={theme}>
  <Router>
    <Header />
  </Router>
<ThemeProvider>
```

The missing ThemeProvider is the reason you see the error `cannot get headerHeight of undefined` in your tests because the `theme` is `undefined`